addOns/ascanrulesBeta/src/test/java/org/zaproxy/zap/extension/ascanrulesBeta/XxeScanRuleUnitTest.java

```diff
@@ -121,4 +129,225 @@ public void replaceMultipleElementsAndRemoveHeader() {
                         + "</comments>";
         assertThat(payload, is(expectedPayload));
     }
+
+    @Test
+    public void shouldScanOnlyIfRequestContentTypeIsXml() throws Exception {
+        // Given
+        String test = "/test";
+        this.nano.addHandler(
+                new NanoServerHandler(test) {
+                    @Override
+                    protected NanoHTTPD.Response serve(NanoHTTPD.IHTTPSession session) {
+                        consumeBody(session);
+                        String response =
+                                "<foo>root:*:0:0:System Administrator:/var/root:/bin/sh</foo>";
+                        return newFixedLengthResponse(
+                                NanoHTTPD.Response.Status.OK, NanoHTTPD.MIME_PLAINTEXT, response);
+                    }
+                });
+
+        HttpMessage msg = this.getHttpMessage(test);
+        msg.setRequestBody("<?xml version=\"1.0\"?><comment><text>test</text></comment>");
+        msg.getRequestHeader().setHeader("Content-Type", "application/json");
+        msg.getRequestHeader().setMethod("POST");
+        rule.init(msg, parent);
+
+        // When
+        rule.scan();
+
+        // Then
+        assertThat(countMessagesSent, equalTo(0));
+    }
+
+    @Test
+    public void shouldSendThreeMessagesForLocalFileReflectionRule() throws Exception {
+        // Given
+        String test = "/test";
+        this.nano.addHandler(
+                new NanoServerHandler(test) {
+                    @Override
+                    protected NanoHTTPD.Response serve(NanoHTTPD.IHTTPSession session) {
+                        consumeBody(session);
+                        String response =
+                                "<foo>root:*:0:0:System Administrator:/var/root:/bin/sh</foo>";
+                        return newFixedLengthResponse(
+                                NanoHTTPD.Response.Status.OK, NanoHTTPD.MIME_PLAINTEXT, response);
+                    }
+                });
+        HttpMessage msg = this.getHttpMessage(test);
+        msg.setRequestBody("<?xml version=\"1.0\"?><comment><text>test</text></comment>");
+        msg.getRequestHeader().setMethod("POST");
+        msg.getRequestHeader().setHeader("Content-Type", "application/xml");
+        rule.init(msg, parent);
+
+        // When
+        rule.localFileReflectionAttack(msg);
```

Comment:
Throughout the new tests the scan method should be used. That ensures none of the payloads are misinterpreted.

However, having the two test methods broken out in the scan rule allows for earlier troubleshooting and maintenance in the future so that was a worthy change (IMHO).

Reply:
Tests have been updated to use the scan method.